### ISSUE TITLE
Add automatic redirect for all /mobile/ app urls to /android/

### ISSUE THREAD

**Issue by EnTeQuAk** [mozilla/addons-frontend#5881]
In #5541 we enabled the user profile but seemed to have missed the `mobile` app for the redirect.

Looking at https://github.com/mozilla-services/cloudops-deployment/blob/master/projects/amo/puppet/modules/amo_proxy/templates/nginx.addons.conf.erb I see that there are quite a few `firefox|android` rules which probably should also add `|mobile` to the expression.

Example: https://addons.mozilla.org/en-US/mobile/user/EnTeQuAk/

@autrilla @bqbn can you take a look at that?

**Comment by bqbn** [comment]
Is it that every `firefox|android` appearance in that file should be changed to `firefox|android|mobile`?

I think historically, we only have `firefox|android` because that's what the app returns when someone visits the `/` endpoint depending on a mobile agent is used or not. 

If the app is now able to return a `mobile` option, we can certainly add that. And just for my information, in what cases will the application return a `<locale>/mobile/` when someone accesses `/`?

**Comment by EnTeQuAk** [comment]
Actually, looking at addons-server code, addons-erver should redirect from `/mobile/` to `/android/` but apparently doesn't do that properly.

Let me check that out first, maybe we don't even need an ops solution.

**Comment by EnTeQuAk** [comment]
Yeah, I updated the subject of this issue, let's rephrase this: Support for the `mobile` app got removed but we should redirect all users to `/android/` automatically. We have some code in addons-server that does that but it seems broken, given that a redirect like that is much more reasonable directly in nginx for performance reasons, can we do that?

Tldr; Redirect all `/{locale}/mobile/` URLs to `/{locale}/android/` since `mobile` doesn't exist anymore (… since 2012 :rofl:)

**Comment by bqbn** [comment]
If `mobile` doesn't exist since 2012, what is the main reason for redirecting `/<locale>/mobile/` to `/<locale>/android/`?

From the access logs, about 2000 out of 52 million requests (~0.003%) are for `/<locale>/mobile/` endpoint yesterday, and the requests are from old browsers such as Gecko/20111104 and etc.

We can add the redirect rule, but the trade-off is that now Nginx needs to check almost 52M more requests to see if that rule matches only because of those 2000 that will match. Does it make sense to let the 2000 see 404 in this case?

**Comment by jvillalobos** [comment]
I think it's fine to show a 404, since `/mobile/` is way past unsupported and the volume is relatively low.

**Comment by muffinresearch** [comment]
Closing based on https://github.com/mozilla/addons-frontend/issues/5881#issuecomment-412136213
